### PR TITLE
[ci] [docker] Up testing to OCaml 4.11.1

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Review by SkySkimmer — DISMISSED** [review]
Didn't update CACHEKEY

File: doc/changelog/11-infrastructure-and-dependencies/12972-ocaml+4_11.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  Coq is tested now against OCaml 4.11.1
```
Comment (Zimmi48) [inline doc/changelog/11-infrastructure-and-dependencies/12972-ocaml+4_11.rst:2]:
```suggestion
  Coq is now tested against OCaml 4.11.1
```

**Comment by ejgallego** [comment]
> Didn't update CACHEKEY

Ups.

**Comment by Zimmi48** [comment]
Pipeline with Docker image build running here: https://gitlab.com/coq/coq/-/pipelines/186947960

**Review by Zimmi48 — CHANGES_REQUESTED** [review]
Please backtrack on the update of ocamlformat. This is an unrelated change, that is not backportable (whereas the rest of the PR would have been) and that would significantly delay the merging of the PR (which was otherwise completely ready).

**Comment by ejgallego** [comment]
> Please backtrack on the update of ocamlformat. This is an unrelated change, that is not backportable (whereas the rest of the PR would have been) and that would significantly delay the merging of the PR (which was otherwise completely ready).

What do you propose, 4.11 is not compatible with ocamlformat 0.14.x

**Comment by ejgallego** [comment]
successful pipeline https://gitlab.com/coq/coq/-/pipelines/186972021

**Comment by Zimmi48** [comment]
> What do you propose, 4.11 is not compatible with ocamlformat 0.14.x , indeed the PR was not ready as not even the docker image completed.

OK then, I was not aware of this and did not look carefully enough at your commit message. Please apply the following change so that the 0.15.0 version is available in `nix-shell` as well:

```diff
diff --git a/dev/nixpkgs.nix b/dev/nixpkgs.nix
index bfb25e72dd..e798645ed0 100644
--- a/dev/nixpkgs.nix
+++ b/dev/nixpkgs.nix
@@ -1,4 +1,4 @@
 import (fetchTarball {
-  url = "https://github.com/NixOS/nixpkgs/archive/17812e653d89c46d68b7b10e290b1c16758f4e47.tar.gz";
-  sha256 = "1zcb70dyfqc8l2ywpbvxmpfshapdi0g365m3rhmwpagqg47pnyxs";
+  url = "https://github.com/NixOS/nixpkgs/archive/3c0e3697520cbe7d9eb3a64bfd87de840bf4aa77.tar.gz";
+  sha256 = "1vx7kyaq0i287dragjgfdj94ggwr3ky2b7bq32l8rkd2k3vc3gl5";
 })
```

**Comment by Zimmi48** [comment]
I'll try to backport this nonetheless, possibly by removing the `ocamlformat` dependency from `v8.12`.

**Comment by ejgallego** [comment]
> I'll try to backport this nonetheless, possibly by removing the `ocamlformat` dependency from `v8.12`.

Indeed, I think that's not a problem for the backport; on the other hand 4.11 didn't see a lot of testing yet with Coq so IMHO we should also tweak the changelog as to suggest the support is experimental or something similar.

Let's see anyways what happens with ocamlformat as if #12985 is merged then no point in keeping it in the docker image.

**Comment by Zimmi48** [comment]
This PR has waited long enough, especially with the 8.12.1 release scheduled for next week. Please apply the patch I gave you and I'll merge.

**Review by Zimmi48 — APPROVED** [review]

**Comment by ejgallego** [comment]
> This PR has waited long enough, especially with the 8.12.1 release scheduled for next week. Please apply the patch I gave you and I'll merge.

Done thanks!

**Comment by Zimmi48** [comment]
Unfortunately that did not work because the nixpkgs version that has ocamlformat 0.15.0 has Sphinx 3, which we are not compatible with yet :disappointed:

**Comment by Zimmi48** [comment]
Fortunately, we are quite close to being compatible with Sphinx 3, so I'll prepare a PR about this.

**Comment by Zimmi48** [comment]
See #13011.

**Comment by Zimmi48** [comment]
OK, this should now work if you rebase the PR

**Comment by Zimmi48** [comment]
ping @ejgallego: this PR just needs a rebase to pass CI

**Comment by ejgallego** [comment]
Done, thanks @Zimmi48

**Comment by Zimmi48** [comment]
@coqbot: merge now

**Pull request merged**